### ISSUE TITLE
Trying to create new model Audiocodes Mediant

### ISSUE THREAD

**Issue by ecm75** [ytti/oxidized#758]
Hi,

me again, this time I'm trying to create a new model for Audiocodes Mediant.
I started by cloning a simple existing model and want to use only ssh.
When connecting to the node oxidize always runs into timeouts without producing any input debug...
I then enabled ssh debug in ssh.rb by adding ```:verbose => :debug``` to ssh_opts and I get this:
```D, [2017-03-07T13:49:39.945331 #17245] DEBUG -- net.ssh.transport.session[1b02d6c]: establishing connection to 192.168.7.17:22
D, [2017-03-07T13:49:39.947466 #17245] DEBUG -- net.ssh.transport.session[1b02d6c]: connection established
I, [2017-03-07T13:49:39.947788 #17245]  INFO -- net.ssh.transport.server_version[1aff978]: negotiating protocol version
D, [2017-03-07T13:49:40.006377 #17245] DEBUG -- net.ssh.transport.server_version[1aff978]: remote is `SSH-2.0-AudioCodes'
D, [2017-03-07T13:49:40.006533 #17245] DEBUG -- net.ssh.transport.server_version[1aff978]: local is `SSH-2.0-Ruby/Net::SSH_3.0.2 x86_64-linux-gnu'
D, [2017-03-07T13:50:46.017408 #17245] DEBUG -- socket[1b02790]: read 0 bytes
D, [2017-03-07T13:50:46.964922 #17245] DEBUG -- net.ssh.transport.session[1bb6ce0]: establishing connection to 192.168.7.17:22
D, [2017-03-07T13:50:46.967232 #17245] DEBUG -- net.ssh.transport.session[1bb6ce0]: connection established
I, [2017-03-07T13:50:46.967534 #17245]  INFO -- net.ssh.transport.server_version[1bb1664]: negotiating protocol version
D, [2017-03-07T13:50:47.007672 #17245] DEBUG -- net.ssh.transport.server_version[1bb1664]: remote is `SSH-2.0-AudioCodes'
D, [2017-03-07T13:50:47.007931 #17245] DEBUG -- net.ssh.transport.server_version[1bb1664]: local is `SSH-2.0-Ruby/Net::SSH_3.0.2 x86_64-linux-gnu'
```
It seems SSH doesn't even get past protocol negotiation.. any ideas?

**Comment by ytti** [comment]
I don't know what it should print next, but what is the actual error you're getting, in logs or crashfile.

**Comment by ecm75** [comment]
Oh, I'm sorry I forgot to tell you it's just timing out.
I think it may be an issue with net::ssh and SSH2. I found similar issues on the web stating that with SSH2 the client should send it's part first... and it seems to be fixed in later versions of net::ssh.
But oxidized seems to require/force an older version of net::ssh.

I put some debug code in ssh.rb and it times out while doing the initial connect and never reaches the login part.

**Comment by ytti** [comment]
You can edit oxidized.gemspec to require net::ssh 4.1 instead and then ```rake build``` to build the package in pkg/ directory, then ```gem install```that

If it helps, we can move to a newer net::ssh version.

**Comment by ecm75** [comment]
Thank you I will give it a try and report back.

**Comment by ecm75** [comment]
It works! ;)
I'm using net/ssh 4.1 now, just had to do some minor modifications to input/ssh.rb to avoid lots of depricated warnings spamming the log. The new net/ssh complains if you pass :proxy a value of nil.

Another note: It seems new models are ignored if they just exists in ~/.config/oxidized/model and oxidized says it can't find that model. Once there is a copy in /var/lib/gems/..../oxidized/model it works and changes to the file in home override the file in /var/lib/...
Should I open a new issue for this?

**Comment by ytti** [comment]
Are you sure you didn't do some mistake? Works fine here with new models _just_ in ~/.config/oxidized/model

**Comment by ecm75** [comment]
I will try to recreate this issue and post some more details...
In my test setup I had a single node in routers.db with the new model, but I have to recreate that to post some messages and dir contents.

**Comment by Fauli83** [comment]
I have tested this today and can verify that using net::ssh 4.1  will solve the problem. Would be great to see this change been default.

**Comment by wk** [comment]
This has been resolved in #909.
